src/foam/u2/filter/StringFilterView.js

```diff
@@ -113,30 +118,19 @@ foam.CLASS({
       required: true
     },
     {
-      name: 'daoContents',
-      preSet: function(_, n) {
-        // remove objects with the same strings for specified property
-        var self = this;
-        return n.reduce(function(accumulator, obj) {
-
-          // create an identifying id
-          var id = obj[self.property];
-
-          // if the id is not found in the temp array
-          // add the object to the output array
-          // and add the key to the temp array
-          if ( accumulator.temp.indexOf(id) === -1 ) {
-            accumulator.out.push(obj);
-            accumulator.temp.push(id);
-          }
-          return accumulator;
-        // return the zero duplicate array
-        }, { temp: [], out: [] }).out;
-      }
+      name: 'daoContents'
     },
     {
       class: 'String',
-      name: 'search'
+      name: 'search',
+      postSet: function(_, n) {
+        this.dao.where(this.CONTAINS_IC(this.property, n)).limit(100).select(this.GroupBy.create({
+          arg1: this.property,
+          arg2: this.ArraySink.create()
```

Comment:
You don't actually use the return values? Just the keys? If so, this could just be a this.COUNT(), which would be faster and send less data across the network.